### PR TITLE
[CMS-1033] add action for building tags and deploying to wp.org

### PR DESCRIPTION
**Pull request by jspellman814**

### CODE REVIEW

**Review by CdrMarks — DISMISSED** [review]
LGTM after the release date is updated.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Contributing
+
+Since 2.3.3 the default branch is `main`. Please make sure you are working against the correct branch. The `master` branch will no longer be accepting pull requests.
+
+## Testing
+
+You may notice there are two sets of tests running, on two different services:
+
+* Travis CI runs the [PHPUnit](https://phpunit.de/) test suite against a Solr instance.
```
Comment (jazzsequence) [inline CONTRIBUTING.md:9]: Can Circle do both? Or GH Actions?

**Review by jabxjab — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+* Circle CI runs the [Behat](http://behat.org/) test suite against a Pantheon site, to ensure the plugin's compatibility with the Pantheon platform.
+
+Both of these test suites can be run locally, with a varying amount of setup.
+
+PHPUnit requires the [WordPress PHPUnit test suite](https://make.wordpress.org/core/handbook/testing/automated-testing/phpunit/), and access to a database with name `wordpress_test`. If you haven't already configured the test suite locally, you can run `bash bin/install-wp-tests.sh wordpress_test root '' localhost`. You'll also need access to a running Solr instance, in order to run the unit tests against Solr.
```
Comment (jazzsequence) [inline CONTRIBUTING.md:14]: > you can run `bash bin/install-wp-tests.sh wordpress_test root '' localhost`.

Is there a script to run this? Can there be?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Behat requires a Pantheon site with Solr enabled. Once you've created the site, you'll need [install Terminus](https://github.com/pantheon-systems/terminus#installation), and set the `TERMINUS_TOKEN`, `TERMINUS_SITE`, and `TERMINUS_ENV` environment variables. Then, you can run `./bin/behat-prepare.sh` to prepare the site for the test suite.
```
Comment (jazzsequence) [inline CONTRIBUTING.md:16]: Perhaps this could be scripted as well?

File: README.md [hunk 4]
```diff
@@ -221,6 +206,9 @@ Add the following to your `functions.php` file.
 
 ## Changelog ##
 
+## 2.3.4-dev (November 17, 2022) ##
```
Comment (jazzsequence) [inline README.md:209]: Was this added automatically? Should it have the `-dev` suffix?
🔢

**Review by jazzsequence — COMMENTED** [review]
Mostly quality of life improvements unrelated to this PR specifically. I am curious about the `-dev` suffix and whether that will get stripped in the release process.

File: README.md (hunk 4 above)
Comment (jspellman814) [reply README.md:209]: I added this with the intent test the `Build and Tag` Action before creating an actual 2.3.4 tag for release.

Comment (jspellman814) [reply README.md:209]: Assuming the CI works as expected, a follow-on PR will be created to remove the `-dev` suffix which will be considered the 2.3.4 release.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jazzsequence) [reply CONTRIBUTING.md:14]: Actually, the above lines are incorrect. Travis isn't being used to run PHPUnit, that's handled by [CircleCI](https://github.com/pantheon-systems/solr-power/blob/2a667f4124a4f04bab037fa8f5e14f8aeb960f97/.circleci/config.yml#L83).

Comment (jspellman814) [reply CONTRIBUTING.md:14]: Adjusted in [026ba64](https://github.com/pantheon-systems/solr-power/pull/551/commits/026ba6496cf8cf1b01e81ea1228b6202b990f6a9)

**Review by jazzsequence — APPROVED** [review]
🚢

**Pull request merged**